homeassistant/components/spotify/media_player.py

```diff
@@ -391,13 +391,18 @@ def update(self) -> None:
         )
         self._currently_playing = current or {}
 
-        context = self._currently_playing.get("context")
+        context = self._currently_playing.get("context", {})
+        uri = context.get("uri")
+        if uri is not None:
+            parts = uri.split(":")
+            if "user" in parts and "playlist" in parts:
+                uri = ":".join([parts[0], parts[3], parts[4]])
```

Comment:
We should not mix searching for items by using `in` with hard coded indices, do one or the other.